packages/bsky/src/api/app/bsky/actor/searchActorsTypeahead.ts

```diff
@@ -82,10 +82,14 @@ const hydration = async (
 }
 
 const noBlocks = (inputs: RulesFnInput<Context, Params, Skeleton>) => {
-  const { ctx, skeleton, hydration } = inputs
-  skeleton.dids = skeleton.dids.filter(
-    (did) => !ctx.views.viewerBlockExists(did, hydration),
-  )
+  const { ctx, skeleton, hydration, params } = inputs
+  skeleton.dids = skeleton.dids.filter((did) => {
+    // Always display exact matches so that users can find profiles that they have blocked
+    const isExactMatch =
+      hydration.actors?.get(did)?.handle?.toLowerCase() ===
+      params.q?.toLowerCase()
```

Comment:
Due to `?.` usage this check will be true if both sides yield undefined/null for some reason, is that intentional?

Reply:
@devinivy not sure what can possibly get returned. i assume as long as we get an actor (added a check for that) it should be okay?